sw/device/lib/base/mmio.h

```diff
@@ -236,6 +236,24 @@ inline void mmio_region_nonatomic_set_mask32(mmio_region_t base,
   mmio_region_write32(base, offset, value);
 }
 
+/**
+ * Sets the bits in `mask` from the MMIO region `base` at the given offset.
+ *
+ * This function is like `nonatomic_set_mask32`, but does not perform a
+ * read, for use with write-only memory.
+ *
+ * @param base the region to mask.
+ * @param offset the offset to apply the mask at, in bytes.
+ * @param mask the mask to set on the selected register.
+ * @param mask_index mask position within the selected register.
+ */
+inline void mmio_region_write_only_set_mask32(mmio_region_t base,
+                                              ptrdiff_t offset, uint32_t mask,
+                                              uint32_t mask_index) {
+  uint32_t value = (mask << mask_index);
```

Comment:
See the main review summary, I think this is duplication, and should be introduced in `bitfield` library that then could be used in `mmio`.